pkg/controller/account/ec2.go

```diff
@@ -0,0 +1,196 @@
+package account
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-logr/logr"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+)
+
+// InitializeSupportedRegions concurrently calls InitalizeRegion to create instances in all supported regions
+// This should ensure we don't see any AWS API "PendingVerification" errors when launching instances
+func (r *ReconcileAccount) InitializeSupportedRegions(reqLogger logr.Logger, regions map[string]map[string]string, creds *sts.AssumeRoleOutput) error {
+	// Create some channels to listen and error on when creating EC2 instances in all supported regions
+	ec2Notifications, ec2Errors := make(chan string), make(chan string)
+
+	// Make sure we close our channels when we're done
+	defer close(ec2Notifications)
+	defer close(ec2Errors)
+
+	for region := range regions {
+		go r.InitializeRegion(reqLogger, region, regions[region]["initializationAMI"], ec2Notifications, ec2Errors, creds)
+	}
+
+	// Wait for all the regions to complete
+	for i := 0; i < len(regions); i++ {
+		select {
+		case msg := <-ec2Notifications:
+			reqLogger.Info(msg)
+		case errMsg := <-ec2Errors:
+			reqLogger.Error(errors.New(errMsg), errMsg)
```

Comment:
This is the point where I mentioned a failed EC2 creation in a region wouldn't be noticed, because we don't return any error.

Reply:
@rogbas we log the error here but we do not fail the account because of it, is that what you mean?

Reply:
Yes, exactly. If we wanted to error out (returning from the function), shoe could consider something like:
```go
		select {
		case msg := <-ec2Notifications:
			reqLogger.Info(msg)
		case errMsg := <-ec2Errors:
			err = errors.New(errMsg)
			reqLogger.Error(err, errMsg)
			return err
		}
```